src/main/java/org/folio/circulation/domain/MoveRequestHelper.java

```diff
@@ -0,0 +1,50 @@
+package org.folio.circulation.domain;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.support.ItemRepository;
+import org.folio.circulation.support.Result;
+
+public class MoveRequestHelper {
```

Comment:
What does this help with?

It appears that it pulls out some dependencies for the move request process and makes it easier to perform some steps.

The `CheckInProcessAdapter` fulfils a similar purpose. Admittedly, this name is far from great.

It'd be nice if we could come up with a descriptive naming convention for these kinds of classes. Any ideas?

Reply:
I took way to long thinking of a relevant name. There was no clear standard naming convention for this type of class. The ultimate goal was to reduce arguments of the move request service, but did provide moving relevant functions into the helper. Honestly, I think an bean registry is ideal for this. We may want to consider Singleton design pattern or possible a factory that can provide all the repositories that require the client.

Reply:
Why do we not pass he clients into the services?

Reply:
If services or relevant classes are stateless, I don’t see why most constructor arguments could not just be clients and request representation if needed.

Reply:
> Why do we not pass the clients into the services?

What do you mean by clients? The class used for creating collection clients?

Reply:
> If services or relevant classes are stateless, I don’t see why most constructor arguments could not just be clients and request representation if needed.

The approach that I've inconsistently applied is to have the constructor declare the actual dependencies and a factory method for default concrete dependencies based upon the clients.

I think using clients for the only constructor starts to hide the dependencies, and makes it harder to eventually move to more unit like tests where these dependencies are faked / mocked.

What do you think?

Reply:
That makes sense. Are there any changes you would like I make here?

Reply:
I think maybe we could shape this to be similar to the `CheckInProcessAdapter`, I'm going to experiment a little